voter/management/commands/voter_process_snapshot.py

```diff
@@ -122,129 +126,139 @@ def reset_file(file_tracker):
     file_tracker.save()
 
 
-@transaction.atomic
+# @transaction.atomic
 def track_changes(file_tracker, output):
     if output:
         print("Tracking changes for file {0}".format(file_tracker.filename), flush=True)
     added_tally = 0
     modified_tally = 0
     ignored_tally = 0
     skip_tally = 0
+    total_lines = 0
 
-    file_date = datetime.strptime(file_tracker.filename.split('/')[-2].split('T', 1)[0], '%Y-%m-%d').replace(tzinfo=pytz.timezone('US/Eastern'))
     voter_records = []
     change_records = []
     voter_updates = []
     processed_ncids = set()
+
+    def flush():
+        with transaction.atomic():
+            if voter_records:
+                NCVoter.objects.bulk_create(voter_records)
+        with transaction.atomic():
+            for c in change_records:
+                c.voter = c.voter
```

Comment:
I saw (but didn't quite understand) your conversation about this in HipChat. Could you add a comment explaining why this is needed?